variables.tf

```diff
@@ -360,11 +360,17 @@ variable "postgres_database_flags" {
 
 ## filstore
 variable filestore_size_in_gb {
-  default = 1024
+  default = 2560
```

Comment:
If `BASIC_HDD` is the default why would this not be `1024` as the doc above states.

Reply:
It was a judgement call. Trying to set a default that works in the largest number of cases.

With a default for 2560, both BASIC_HDD and BASIC_SSD will work by default.
With a default of 1024 BASIC_SDD will fail unless you also increase filestore_size.

Note that CONFIG_VARS.md states that you can reduce the filesize for basic_sdd. 
In practice, it's not a big cost factor (~$20 per TB per month), and just as well to have plenty storage.

Reply:
Can we not put a check in there to guard against that if it's HDD > 1024 if it's SDD > 2560?

Reply:
We can put a check in. It's not really necessary, though. The Filestore will kick you out with a reasonable error message, too.
I was just aiming for a smoother user experience (i.e. avoid cascading dependencies between parms)

Reply:
How about in locals:

```
filestore_size_in_gb = var.filestore_tier == "BASIC_HDD" ? (var.filestore_size_in_gb < 1024 ? 1024 : var.filestore_size_in_gb) : (var.filestore_size_in_gb < 2560 ? 2560 : var.filestore_size_in_gb)
```

And then set the default value to `null` this way if the user chooses a value its checked and set according to the `filestore_tier` selection.

Reply:
The reasoning for the logic I provided above is that it will not let the user set illegal values. So yes it does overwrite what they may have, but only to protect the system from throwing an error. The code you have will simply allow them to set a value without any error checking. Which does cause an error which can be prevented.

Reply:
I see your reasoning. But I think it runs up against a hard principle of never overriding user-inputs.
Say a user specifies 500g - ignoring all documentation (it does happen!) - and the "terraform apply" will succeed. 
The value set by user does  now not match the infrastructure, the project did not create the infrastructure that the user explicitly requested.
Later the user notices the 1T value in the actual Filestore and will be confused at best, and raise a bug at worst.
 
Better to stop processing and let users know when they set invalid values.

Reply:
I kinda agree, but if you tell someone that the minimum value is x and anything less is not accepted. To have them enter x and throw and error seems contrary. I believe the user is simply trying to create a usable system. My thought is that if they ignore the docs, they get what we give them. Everyone should read the docs, especially on a project like this one. But I'll let it go for now. If you can implement your simplified code.